randovania/games/am2r/gui/preset_settings/am2r_teleporters_tab.py

```diff
@@ -0,0 +1,144 @@
+from randovania.layout.lib.teleporters import (
+    TeleporterConfiguration,
+    TeleporterList,
+    TeleporterShuffleMode,
+)
+
+if TYPE_CHECKING:
+    from PySide6 import QtWidgets
+
+    from randovania.game_description.db.area import Area
+    from randovania.game_description.db.node_identifier import NodeIdentifier
+    from randovania.layout.preset import Preset
+
+
+class PresetTeleportersAM2R(PresetTeleporterTab, Ui_PresetTeleportersAM2R, NodeListHelper):
+    teleporter_mode_to_description = {
+        TeleporterShuffleMode.VANILLA: "All transport pipes are connected to where they do in the original game.",
+        TeleporterShuffleMode.TWO_WAY_UNCHECKED: (
+            "After taking a transport pipe, the transport pipe in the room you are in"
+            " will bring you back to where you were. Transport pipes can connect to another one in the same region."
+        ),
+        TeleporterShuffleMode.ONE_WAY_TELEPORTER: (
+            "All transport pipes bring you to a transport pipe room, but going backwards can go somewhere else. "
+            "All rooms are used as a destination exactly once, causing all transport pipes to be separated into loops."
+        ),
+        TeleporterShuffleMode.ONE_WAY_TELEPORTER_REPLACEMENT: (
+            "All transport pipes bring you to a transport pipe room, but going backwards can go somewhere else. "
+            "Rooms can be used as a destination multiple times, causing transport pipes which you can possibly"
+            " not come back to."
+        ),
+    }
+
+    def setup_ui(self) -> None:
+        self.setupUi(self)
+
+    @classmethod
+    def tab_title(cls) -> str:
+        return "Transport Pipes"
+
+    def _create_source_teleporters(self) -> None:
+        row = 0
+        region_list = self.game_description.region_list
+
+        locations = TeleporterList.nodes_list(self.game_enum)
+        node_identifiers: dict[NodeIdentifier, Area] = {
+            loc: region_list.area_by_area_location(loc.area_identifier) for loc in locations
+        }
+        checks: dict[NodeIdentifier, QtWidgets.QCheckBox] = {
+            loc: self._create_check_for_source_teleporters(loc) for loc in locations
+        }
+
+        self._teleporters_source_for_location = copy.copy(checks)
+        self._teleporters_source_destination: dict[NodeIdentifier, NodeIdentifier | None] = {}
+
+        for location in sorted(locations, key=lambda loc: (0, checks[loc].text())):
+            if location not in checks:
+                continue
+
+            self.teleporters_source_layout.addWidget(checks.pop(location), row, 1)
+
+            other_locations = [
+                node.default_connection
+                for node in node_identifiers[location].nodes
+                if isinstance(node, DockNode)
+                and node.dock_type in self.teleporter_types
+                and region_list.identifier_for_node(node) == location
+            ]
+            assert len(other_locations) == 1
```

Comment:
I'm not sure, i just copy pasted this from dread xP
Looking at it, I have no idea what it was trying to do. It may have been trying to check if a `default_connection` exists, but even if that one's None, it'd still get added to the list.
Either way, this is completely redundant now, so I removed it.